src/node/mini_miner.h

```diff
@@ -103,13 +109,28 @@ class MiniMiner
     bool IsReadyToCalculate() const { return m_ready_to_calculate; }
 
     /** Build a block template until the target feerate is hit. */
-    void BuildMockTemplate(const CFeeRate& target_feerate);
+    void BuildMockTemplate(std::optional<CFeeRate> target_feerate);
 
     /** Returns set of txids in the block template if one has been constructed. */
     std::set<uint256> GetMockTemplateTxids() const { return m_in_block; }
 
+    /** Constructor that takes a list of outpoints that may or may not belong to transactions in
+     * mempool. */
     MiniMiner(const CTxMemPool& mempool, const std::vector<COutPoint>& outpoints);
 
+    /** Constructor in which MiniMinerMempoolEntry entries have been constructed manually,
+     * presumably because these transactions are not in the mempool (yet). It is assumed that all
+     * values are correct, otherwise MiniMiner algorithms won't work. Callers should check
```

Comment:
I think it can precise what is yield by “MiniMiner algorithms won’t work” e.g calculate non-optimal mining scores.

Reply:
I've added "otherwise the results may be incorrect" if that works?